### ISSUE TITLE
Plugin install fails with "No such file..."

### ISSUE THREAD

**Issue by peske** [elastic/puppet-elasticsearch#643]
I've tried installing [dejaVu](https://github.com/appbaseio/dejaVu) plugin by using `elasticsearch::plugin` class, but it has ended with the error:

```
No such file or directory @ rb_sysopen - /usr/share/elasticsearch/plugins/dejavu/.name
```

At first I've thought that it is about the plugin itself, and wanted to report the problem there, but when I've tried to install the plugin directly, without puppet, it was successful:

```
Installed dejaVu into /usr/share/elasticsearch/plugins/dejaVu
```

So obviously the problem is with puppet-elasticsearch module. Is it a bug? Or I'm doing something wrong?

Thanks!

**Comment by tylerjl** [comment]
Interesting issue! Thanks for reporting.

I actually think I know what the problem is - when puppet manages the plugin, it creates the `.name` file to track the plugin metadata for puppet to use later. Based upon what I'm seeing, it may actually be trying to read it in the `dejavu` directory (note the case) but the plugin is installed within a directory with the fancy `dejaVu` case you're seeing.

Do you have information regarding the version of the module you're using, version of puppet, OS and version, and the puppet plugin resource in the manifest you're using to install dejaVu to see if I can reproduce it?

**Comment by peske** [comment]
You've probably right that it is about the case. Regarding my environment: pretty much newest in everything:

- ElasticSearch 2.3 (or 2.3.3? the latest definitely since I've installed it yesterday)
- ElasticSearch Module 0.11.0
- Puppet Agent 4.5.1
- Puppet Master 4.5.1
- Both ElasticSearch and Puppet server are running on Ubuntu 14.04.4
- dejaVu Module - the latest again

My manifest:

```
elasticsearch::plugin { 'appbaseio/dejaVu':
  instances  => $node_name,
  proxy_host => 'internal-proxy.itenlight.com',
  proxy_port => 8080, 
}
```

`$node_name` is `IT-ES01`. I've installed several more plugins without problem:

```
elasticsearch::plugin { 'lmenezes/elasticsearch-kopf':
  instances  => $node_name,
  proxy_host => 'internal-proxy.itenlight.com',
  proxy_port => 8080, 
}

elasticsearch::plugin { 'mobz/elasticsearch-head':
  instances  => $node_name,
  proxy_host => 'internal-proxy.itenlight.com',
  proxy_port => 8080, 
}
```

**Comment by bisscuitt** [comment]
I'm having this same issue, but with a differetn module. It seems that the module dir is not being correctly auot detected.

```
Error: Could not set 'present' on ensure: No such file or directory @ rb_sysopen - /usr/share/elasticsearch/plugins/swift-repository-plugin/.name at 146:/opt/puppet_envs/es_fix_snapshots/librarian-modules/elasticsearch/manifests/plugin.pp
Error: Could not set 'present' on ensure: No such file or directory @ rb_sysopen - /usr/share/elasticsearch/plugins/swift-repository-plugin/.name at 146:/opt/puppet_envs/es_fix_snapshots/librarian-modules/elasticsearch/manifests/plugin.pp
Wrapped exception:
No such file or directory @ rb_sysopen - /usr/share/elasticsearch/plugins/swift-repository-plugin/.name
Error: /Stage[main]/Bigcommerce_elasticsearch::Plugins/Bigcommerce_elasticsearch::Plugin[org.wikimedia.elasticsearch.swift/swift-repository-plugin/2.3.3.1]/Elasticsearch::Plugin[org.wikimedia.elasticsearch.swift/swift-repository-plugin/2.3.3.1]/Elasticsearch_plugin[org.wikimedia.elasticsearch.swift/swift-repository-plugin/2.3.3.1]/ensure: change from absent to present failed: Could not set 'present' on ensure: No such file or directory @ rb_sysopen - /usr/share/elasticsearch/plugins/swift-repository-plugin/.name at 146:/opt/puppet_envs/es_fix_snapshots/librarian-modules/elasticsearch/manifests/plugin.pp
```

The plugin is `org.wikimedia.elasticsearch.swift/swift-repository-plugin`
```
elasticsearch::plugin { 'org.wikimedia.elasticsearch.swift/swift-repository-plugin/2.3.3.1': 
      ensure => present, 
}
```

Based on the name of the plugin, the puppet module is expecting the module dir to be `swift-repository-plugin` however, the dir created after the install is `swift-repository`:

```
# ls -la /usr/share/elasticsearch/plugins/
total 20
drwxr-xr-x  5 elasticsearch elasticsearch 4096 Jul 14 22:48 .
drwxr-xr-x 10 elasticsearch elasticsearch 4096 Jun 22 02:24 ..
drwxr-xr-x  2 root          root          4096 Jul 12 00:11 delete-by-query
drwxr-xr-x  8 root          root          4096 Jun 20 21:42 kopf
drwxr-xr-x  2 root          root          4096 Jul 14 22:48 swift-repository
```

The other plugins `kopf` and `delete-by-query` have installed correctly via the puppet module.

System:
* ElasticSearch 2.3.3
* ElasticSearch Module 0.11.0

With older version of this module we were able to set `module_path` to override the path to the plugin.

**Comment by tylerjl** [comment]
With the fix in #660 from @bisscuitt and original fix in #646, these issues should be resolved and, at the very least, overrideable via the `module_dir` option again. Should be a part of the 0.13.0 release. Thanks for the PRs and feedback, y'all! 👍
